### ISSUE TITLE
Symbol lookup error after upgrading

### ISSUE THREAD

**Issue by CommanderKronos** [Alexays/Waybar#469]
Hi there. I'm using waybar on arch. I recently upgraded using pacman -Syu. And since then I got the error.
`waybar: symbol lookup error: waybar: undefined symbol: _ZN4Json5ValueaSES0_`

steps to reproduce
run waybar from terminal

(This is my first ever issue i open on github. So if I'm supposed to add anything or do something else please let me know)

**Comment by ppascher** [comment]
Not sure if it is the same issue but since you mentioned using arch: What is your libjsoncpp version? Mine was updated to 1.9.1-1 about a week ago and it does not provide the libjsoncpp.so.20 required by Waybar anymore but .21 instead. I tried a symbolic link but that does not work unfortunatly. I get a `waybar: error while loading shared libraries: libjsoncpp.so.20: cannot open shared object file: No such file or directory` whet trying to start waybar. A downgrade should work.

**Comment by CommanderKronos** [comment]
Ow okay! But I don't actually know how I can downgrade. Do you know how I could do that?

**Comment by CommanderKronos** [comment]
Hey! I fixed my issue. Like ppascher said 

> A downgrade should work.

Just downgrade jsoncpp to 1.8.4 and waybar should work again. To downgrade I used the downgrader tool: https://github.com/pbrisbin/downgrade
I had to remove waybar though. Because this tool gives an error when it encounters dependencies. 

Steps I used to fix:
-Remove waybar
-Downgrade jsoncpp
-Recompile and make waybar
-Profit

Should I close this issue now? I don't know how github works yet.

**Comment by gnxlxnxx** [comment]
Yet another lookup error, not sure either if that's the same issue:

`waybar: symbol lookup error: /usr/lib/libspdlog.so.1: undefined symbol: _ZN3fmt2v68internal14sprintf_formatIdEEPcT_RNS1_6bufferIcEENS1_13sprintf_specsE`

I'm going to have to downgrade too then...

**Comment by gnxlxnxx** [comment]
Yet another lookup error, not sure either if that's the same issue:

`waybar: symbol lookup error: /usr/lib/libspdlog.so.1: undefined symbol: _ZN3fmt2v68internal14sprintf_formatIdEEPcT_RNS1_6bufferIcEENS1_13sprintf_specsE`

I'm going to have to downgrade too then...

EDIT: doesn't really seem to have to do anything specifically with waybar saw the issue with other programs as well probably a issue with a dependency...

**Comment by FreeFull** [comment]
That's an issue with fmt. Upgrade fmt to 6.1.2 or downgrade to 6.1.0

**Comment by gnxlxnxx** [comment]
Oh great thanks my aur helper (pacaur) has somehow not upgraded that.

**Comment by ppascher** [comment]
I just took a look again and noticed the 'waybar' command I was executing was located under '/usr/local/bin' while the current build (using arch's latest jsoncpp) installed to '/usr/bin'. Using the correct path fixes this issue for me. Sorry for the noise.

**Comment by guanghwang** [comment]
It fixes the problem, using @ppascher method. Not sure why I have waybar in two locations.

**Comment by dmitridb** [comment]
I'm finding this as an issue with latest version of fmt again.

```pacman -U /var/cache/pacman/pkg/fmt-7.0.3-1-x86_64.pkg.tar.zst```

and boom fixed

**Comment by SweetestRug** [comment]
I am experiencing the same problem on Manjaro (which it looks inherited the issue from Arch). Downgrading to fmt 7.0.3-1 solves the problem...for now.

**Comment by diraol** [comment]
I'm facing a smiliar issue:

> waybar: symbol lookup error: waybar: undefined symbol: _ZN6spdlog7details7log_msgC1ENS_10source_locEN3fmt2v617basic_string_viewIcEENS_5level10level_enumES6_

I'm using debian:

> No LSB modules are available.
> Distributor ID:	Debian
> Description:	Debian GNU/Linux bullseye/sid
> Release:	unstable
> Codename:	sid

And waybar installed from debian repo:

> waybar/unstable,now 0.9.4-1+b1 amd64

Since there were some comments about libfmt, here is the version I have:

> libfmt7/unstable,now 7.1.3+ds1-4 amd64

And also:

> libjsoncpp24/unstable,now 1.9.4-4 amd64

Any suggestion on how to debug / figure out what is going on is very welcome!

**Comment by dmitridb** [comment]
I couldn't get a thing with this to compile and run on Debian or Ubuntu with an up to date version of sway. Best advice is to reconsider your choice of distro for its package management philosophy if you want to use this without headaches, unless managing sid backports that maybe don't even exist along with the current version of waybar sounds like fun to you. The supporting libs packages with the distro are simply too crumbly and old for moving targets like this without someone supporting it on the distro side
-------- Original Message --------

**Comment by dmitridb** [comment]
wait you're actually running sid? yeah might be fixed in some days on package maintainers end but still having to deal with this on a distro like that was more trouble than it was worth for me personally.
-------- Original Message --------

**Comment by diraol** [comment]
Yeap, I'm a `sid` user! (and I'm pretty much aware of what does this mean).

To be honest, usually it is quite ok. Indeed sometimes I face some issues that are fixed within some days (usually less then a day). And I'm good with it, especially because debian is the distro I feel more comfortable with.

Anyway, if you guys have tips for me in order to track down the error so I can file a bug report (in the broken package tracker) that would be welcome :)

**Comment by dmitridb** [comment]
if you can figure out how apt does an operation like pacman -U in arch, you're good. I just don't know if that exists, I try and run stable personally because that's debian's stronger point... and mostly just on servers
-------- Original Message --------

**Comment by diraol** [comment]
What does `pacman -U` does?

Regarding stables and servers, then I totally agree, for servers I would never think about sid!

**Comment by dmitridb** [comment]
allows you to downgrade to a specific recent version in the package cache
-------- Original Message --------

**Comment by diraol** [comment]
Hum... I just found out something weird.

I have another computer with debian where I have a mix of testing and unstable packages (yeah, I know, this goes against debian recommendations).

In this second computer I the same waybar version installed, but I don't have the `libfmt7` package installed, and waybar is working perfectly.

Eventually this has something to do with the fact that I had built waybar before it was available with `apt`.

I'll try to install `fmt` from source and see what happens.

Regarding downgrades, we can `apt install <packager>=<version>`. But that requires the version to be available and, in this specific case, neither waybar nor libfmt7 are available in different versions other than the one installed. (I don't know debian policies for package versions in sid)
